tests/AnalyzeSpec.hs

```diff
@@ -1142,15 +1142,15 @@ spec = describe "analyze" $ do
   describe "pact-id" $ do
     let code =
           [text|
-            (defun test:integer ()
+            (defun test:string ()
               (pact-id))
           |]
 
     expectPass code $ Satisfiable Abort'
     expectPass code $ Satisfiable Success'
-    expectPass code $ Satisfiable $ CoreProp $ IntegerComparison Eq
-      (Inj Result :: Prop 'TyInteger)
-      10
+    expectPass code $ Satisfiable $ CoreProp $ StrComparison Eq
+      (Inj Result :: Prop 'TyStr)
+      (Lit' "tx-id:1:chain-id:" :: Prop 'TyStr)
```

Comment:
This doesn't return false even when the comparison string is obviously wrong (i.e. `(Lit' "wrong-pact-id" :: Prop 'TyStr)`).

Reply:
see above, moving this to hash

Reply:
Ideally, once generated, pact code shouldn't care what a pact-id is as it's not a specific datatype ...

Reply:
Changed to a hash.